skipper_test.go

```diff
@@ -641,3 +642,139 @@ func createFilterRegistry(specs ...filters.Spec) filters.Registry {
+	fdApp, err := os.CreateTemp("/tmp", "app_log_")
+	if err != nil {
+		t.Fatalf("Failed to create tempfile: %v", err)
+	}
+	defer fdApp.Close()
+
+	// access log
+	fdAccess, err := os.CreateTemp("/tmp", "access_log_")
+	if err != nil {
+		t.Fatalf("Failed to create tempfile: %v", err)
+	}
+	defer fdAccess.Close()
+
+	// run skipper proxy that we want to test
+	o := Options{
+		Address:                         ":8090",
+		EnableRatelimiters:              true,
+		SourcePollTimeout:               1500 * time.Millisecond,
+		WaitFirstRouteLoad:              true,
+		SuppressRouteUpdateLogs:         false,
+		MetricsListener:                 ":8091",
+		testOptions:                     testOptions{},
+		RoutesFile:                      filePath,
+		InlineRoutes:                    `healthz: Path("/healthz") -> status(200) -> inlineContent("OK") -> <shunt>;`,
+		ApplicationLogOutput:            fdApp.Name(),
+		AccessLogOutput:                 fdAccess.Name(),
+		AccessLogDisabled:               false,
+		MaxTCPListenerConcurrency:       0,
+		ExpectedBytesPerRequest:         1024,
+		ReadHeaderTimeoutServer:         0,
+		ReadTimeoutServer:               1 * time.Second,
+		MetricsFlavours:                 []string{"codahale"},
+		EnablePrometheusMetrics:         true,
+		LoadBalancerHealthCheckInterval: 3 * time.Second,
+		OAuthTokeninfoURL:               "http://127.0.0.1:12345",
+		CredentialsPaths:                []string{"/does-not-exist"},
+		EnableSwarm:                     true,
+		SwarmPort:                       9001,
+		CompressEncodings:               []string{"gzip"},
+		IgnoreTrailingSlash:             true,
+		EnableBreakers:                  true,
+		DebugListener:                   ":8092",
+		StatusChecks:                    []string{"http://127.0.0.1:8091/metrics", "http://127.0.0.1:8092"},
+	}
+
+	dcs, err := createDataClients(o, innkeeper.CreateInnkeeperAuthentication(innkeeper.AuthOptions{}), nil)
+	if err != nil {
+		t.Fatalf("Failed to createDataclients: %v", err)
+	}
+
+	fr := createFilterRegistry(
+		fscheduler.NewFifo(),
+		flog.NewEnableAccessLog(),
+		builtin.NewStatus(),
+		builtin.NewInlineContent(),
+	)
+	metrics := &metricstest.MockMetrics{}
+	reg := scheduler.RegistryWith(scheduler.Options{
+		Metrics:                metrics,
+		EnableRouteFIFOMetrics: true,
+	})
+	defer reg.Close()
+
+	// create LB in front of apiservers to be able to switch the data served by apiserver
+	ro := routing.Options{
+		SignalFirstLoad: true,
+		FilterRegistry:  fr,
+		DataClients:     dcs, //[]routing.DataClient{dc},
+		PostProcessors: []routing.PostProcessor{
+			loadbalancer.NewAlgorithmProvider(),
+			reg,
+		},
+		SuppressLogs: true,
+	}
+	rt := routing.New(ro)
+	defer rt.Close()
+	<-rt.FirstLoad()
+	tracer := &tracingtest.Tracer{}
+	pr := proxy.WithParams(proxy.Params{
+		Routing:     rt,
+		OpenTracing: &proxy.OpenTracingParams{Tracer: tracer},
+	})
+	defer pr.Close()
+	lb := stdlibhttptest.NewServer(pr)
+	defer lb.Close()
```

Comment:
I saw a flake
```
--- FAIL: TestDataClients (1.01s)
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:527: Waiting for proxy being ready
    skipper_test.go:538: Failed to GET routes file route: Get "http://localhost:8090/routes-file": dial tcp 127.0.0.1:8090: connect: connection refused
```

TBH I can not understand what this is testing.

Reply:
That skipper starts with safety nets enabled like status check.